src/sentry/api/helpers/group_index/index.py

```diff
@@ -77,6 +79,34 @@ def build_query_params_from_request(
         except ValueError:
             raise ParseError(detail="Invalid cursor parameter.")
     query = request.GET.get("query", "is:unresolved").strip()
+    if request.GET.get("savedSearch") == "0":
+        saved_searches = (
+            SavedSearch.objects
+            # Do not include pinned or personal searches from other users in
+            # the same organization. DOES include the requesting users pinned
+            # search
+            .exclude(
+                ~Q(owner_id=request.user.id),
+                visibility__in=(Visibility.OWNER, Visibility.OWNER_PINNED),
+            )
+            .filter(
+                Q(organization=organization) | Q(is_global=True),
+            )
+            .extra(order_by=["name"])
+        )
+        selected_search_id = request.GET.get("searchId", None)
+        if selected_search_id:
+            # saved search requested by the id
+            saved_search = saved_searches.filter(id=int(selected_search_id))
+        else:
+            # pinned saved search
+            saved_search = saved_searches.filter(visibility=Visibility.OWNER_PINNED)
+
+        if saved_search.exists():
+            search = saved_search.first()
+            query_kwargs["sort_by"] = search.sort
+            query = search.query
```

Comment:
What does this query for if the saved_search isn't found?

Reply:
yup, it'll use the default is:unresolved query